### PR TITLE
WebUI: Update Check, Cleanup and Hyperion Switcher IPv6 Fix

### PR DESCRIPTION
**Pull request by b1rdhous3**
Comments welcome. Especially on the update and tag process. 

**1.** Tell us something about your changes. 

Update Check:
- New Property which allows you to select stable or beta branch checks.
- The last stable will be shown as new version when the latest stable version have a higher version number than the last beta version. This only triggers when Beta Branch is used. If the user chooses Stable only stable versions will be considered.
- This is based on GitHub api which gets release list. Newest release of selected branch will be shown at dashboard. Also implemented updates.html again with the right logic. 
**- If we publish one beta here you can see first results of this.** 

I would suggest to make beta builds available under releases to. Then our install script can search for newest beta or stable tag (depending on what the user wants) and update this way. 

Cleanup:
- I cleaned up the WebUI from dead links and the support page with the affiliate links and donation adresses as we don't have them under our control. Currently only disabled, not deleted

What are your thoughts about this? Should we delete this parts for now?

Switcher:
- If the address which is published via bonjour serve is IPv6 the Switcher was bugged. Fixed that.

**2.** If this changes affect the .conf file. Please provide the changed section

"general" :
	{
		"name"       : "My Hyperion Config",
		"versionBranch" : "Stable",
		"showOptHelp" : true
	}

**3.** Reference an issue (optional)

Note: For further discussions use our forum: forum.hyperion-project.org

### CODE REVIEW

**Comment by b1rdhous3** [comment]
Not ready to merge now.

**Comment by b1rdhous3** [comment]
I am ready for merge :)

File: assets/webconfig/js/content_dashboard.js [hunk 1]
```diff
@@ -78,20 +78,43 @@ $(document).ready( function() {
...
-		window.parsedUpdateJSON = JSON.parse(data);
-		window.latestVersion = window.parsedUpdateJSON[0].versionnr;
-	//	var cleanLatestVersion = window.latestVersion.replace(/\./g, '');
-	//	var cleanCurrentVersion = window.currentVersion.replace(/\./g, '');
+	$.get( window.gitHubReleaseApiUrl, function( releases ) {
+		window.gitHubVersionList = releases;
 
-		$('#dash_latev').html(window.currentVersion);
-	//	$('#dash_latev').html(window.latestVersion);
+		for(i in releases)
```
[the remark this answers is not in the document]
Comment (b1rdhous3) [reply assets/webconfig/js/content_dashboard.js]: Done.

**Comment by Paulchen-Panther** [comment]
I have the problem when I open the WebUI and go to the entry "Update" after updating the page (F5), the WebUI not fully loaded. It sticks to the Hyperion logo. Could you test that again please?
Could you catch this error message?
test

**Comment by b1rdhous3** [comment]
I see. I have rewritten the code for getting Releases. It now uses a callback help function. Also the error with the stuck logo will be fixed in the next commit. 
Sadly you can not prevent this error message at 404. But I can catch it and handle further steps. I will do this in my callback function. But the message with 404 will remain.
Please see https://stackoverflow.com/questions/16481598/prevent-unhandled-jquery-ajax-error

**Comment by b1rdhous3** [comment]
Ok, you can test again.

**Comment by Paulchen-Panther** [comment]
Jetzt haben wir eine Fehlermeldung mehr. Dafür bleibt das WebUI nicht hängen. :+1: 
Bildschirmfoto vom 2019-06-26 18-51-10

**Comment by b1rdhous3** [comment]
Hm der Fehler kam bei mir bisher nicht. Kommt direkt beim initialen öffnen des Dashboards? 
Schaue ich mir an.

**Comment by Paulchen-Panther** [comment]
> Hm der Fehler kam bei mir bisher nicht. Kommt direkt beim initialen öffnen des Dashboards?
> Schaue ich mir an.

Ja

**Comment by b1rdhous3** [comment]
nun aber

**Comment by Paulchen-Panther** [comment]
Einwandfrei. Hast du noch was oder soll ich zusammenführen?

**Comment by b1rdhous3** [comment]
@Paulchen-Panther kannst zusammenführen

@brindosch I hope. We have to think about how to start the update. I think we could send an command with the version tag in it to hyperiond which will trigger our installation/update script.

**Pull request merged**